silverback/runner.py

```diff
@@ -55,6 +100,23 @@ async def run(self):
         Raises:
             :class:`~silverback.exceptions.Halt`: If there are no configured tasks to execute.
         """
+        if settings.MONGODB_URI:
+            self.persistence = await init_mongo(settings.MONGODB_URI)
+
+        if self.persistence:
+            boot_state = await self.persistence.get_instance_state(self.ident)
+            if boot_state:
+                self.last_block_seen = boot_state.last_block_seen
+                self.last_block_processed = boot_state.last_block_processed
+
+            # NOTE: This will update TaskIQ state before app startup.
+            # TODO: I have no idea if this will work with a distributed runner/worker
+            # setup.  That may require our own injected startup task, but it's unclear if
+            # we can do that before Taskiq's startup handlers.  We might be able to force
+            # SilverbackApplication to decorate an internal func to handle it.  Need to
+            # figure that out.
+            self.app.checkpoint(self.last_block_seen, self.last_block_processed)
```

Comment:
I have no idea if state changes are communicated between client and workers.  I assume not.  I bet it just happens that InMemoryBroker uses a singleton value for client and workers.  If that's the case, this isn't going to work and we need to setup some kind of communication to allow the runner to update the workers.

Will need to figure out how to run Silverback with a different broker so I can get a local env to test with.

Reply:
This was bunk.  Removed.